spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/core/convert/StructAccessor.java

```diff
@@ -125,6 +126,9 @@ Object getSingleValue(String colName) {
 	}
 
 	public Object getSingleValue(int colIndex) {
+		if (this.struct.isNull(colIndex)) {
```

Comment:
In what situation would this be `true`? Can you contrive a test for it?

Reply:
This is kind of unrelated to json support. But when I was testing, I found an edge case when the query result is a simple field, and return is null, it fails. Like this one. 
`@Query("SELECT name from test where id = @id")`
`Optional<String> getNameById(@Param("id") String id);`

I'll add it to a test somewhere.

Reply:
Added a test [here](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/0e4aae3dfba729a10295f12604937953646dbc66/spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/repository/it/SpannerRepositoryIntegrationTests.java#L115-L120).

Reply:
filed issue: #637.